### PR TITLE
Add support to pass in TLS key/crt to use gRPC securely

### PR DESCRIPTION
**Pull request by IvanTheGreatDev**
This is for #366

It originally said to use gRPC securely by default but that would be unfeasible since node's need their own certs/keys.

### CODE REVIEW

File: client/rpcclient/service.go [hunk 1]
```diff
@@ -30,13 +32,26 @@ func NewRPCClient(ctx context.Context, cfg *Config) *Service {
...
+		withCert: cfg.CertFlag,
 	}
 }
 
 // Start the grpc connection.
 func (s *Service) Start() {
 	log.Info("Starting service")
-	conn, err := grpc.Dial(s.endpoint, grpc.WithInsecure())
+	var server grpc.DialOption
+	if s.withCert != "" {
```
Comment (terenc3t) [inline client/rpcclient/service.go]: use nil?

File: client/rpcclient/service.go [hunk 2]
```diff
@@ -30,13 +32,26 @@ func NewRPCClient(ctx context.Context, cfg *Config) *Service {
...
+		creds, err := credentials.NewClientTLSFromFile(s.withCert, "")
+		if err != nil {
+			log.Errorf("Could not get valid credentials: %v", err)
+			return
+		}
+		server = grpc.WithTransportCredentials(creds)
+	} else {
+		server = grpc.WithInsecure()
+		log.Warnf("Your gRPC connection is insecure!")
```
Comment (terenc3t) [inline client/rpcclient/service.go]: don't need format option here. can you just use `log.warn`?

File: beacon-chain/rpc/service.go [hunk 3]
```diff
@@ -47,7 +54,20 @@ func (s *Service) Start() {
 	s.listener = lis
 	log.Infof("RPC server listening on port :%s", s.port)
 
-	grpcServer := grpc.NewServer()
+	var grpcServer *grpc.Server
+	if s.withCert != "" && s.withKey != "" {
```
[the remark this answers is not in the document]
Comment (IvanTheGreatDev) [reply beacon-chain/rpc/service.go]: `s.withCert` is a string so it fails

[the remark this answers is not in the document]
Comment (terenc3t) [reply beacon-chain/rpc/service.go]: I see. In that case use `ctx.GlobalIsSet(utils.CertFlag.Name)`

[the remark this answers is not in the document]
Comment (IvanTheGreatDev) [reply beacon-chain/rpc/service.go]: Is there access to that from `Start()`? I'm trying to use `s.ctx` and it doesn't have `GlobalIsSet`.

File: client/rpcclient/service.go (hunk 1 above)
Comment (IvanTheGreatDev) [reply client/rpcclient/service.go]: This is the same problem as the previous, s.withCert is a string so it cannot be compared with nil.

Comment (terenc3t) [reply client/rpcclient/service.go]: no worries, I'm looking if there's a better way to pass context from node level to service level

**Comment by terenc3t** [comment]
Lint failed

**Pull request merged**